### PR TITLE
Restructure side menu to allow for subheadings.

### PR DESCRIPTION
**Pull request by dmfalke**
Related to #241

This PR contains preliminary work to add the collections map type to the eda map.

Changes include:

- Variable renaming
- Modifications to finding active menu item and active map type
- Extracting some components

_Note that this PR is targeting the branch `map-collection-variable-map-type`, rather than `main`. The intent is to allow for incremental PR merges for #241. We'll see how that goes._

### CODE REVIEW

**Comment by bobular** [comment]
I see that `map-collection-variable-map-type` doesn't have its own PR yet, which is fine.

But there seems to be some funny business going on in github:

image

**Comment by jernestmyers** [comment]
> I see that `map-collection-variable-map-type` doesn't have its own PR yet, which is cool with me. Though perhaps it could have a draft PR?
> 
> But there seems to be some funny business going on in github:
> 
> image

This is very weird. I haven't even checked out the branch yet!

**Review by jernestmyers — APPROVED** [review]
LGTM. Haven't fully absorbed all the changes, but it does appear like a more mature version. I like the improvements to the typing and naming. I imagine it's more readable to folks that haven't worked with these components in the past. Having recently worked with these components, there's enough memory that there's a bit of a cognitive dissonance going on with all the change- but I think it's an improvement for sure!

File: packages/libs/eda/src/lib/map/analysis/MapSideNavigation.tsx [hunk 1]
```diff
@@ -1,190 +1,121 @@
...
-  activeNavigationMenu,
-  children,
-  isExpanded,
-  onToggleIsExpanded,
-  siteInformationProps,
-}: MapSideNavigationProps) {
-  const sideMenuExpandButtonWidth = 20;
+  function formatEntry(entry: SidePanelMenuEntry, isActive = false) {
+    const style = {
+      paddingLeft: `${indentLevel * 1.3 + 0.5}em`,
```
[the remark this answers is not in the document]
Comment (dmfalke) [reply packages/libs/eda/src/lib/map/analysis/MapSideNavigation.tsx:27]: Good question! For each indent level, except for the top-most, we want to add 1.3em of padding to the left. Then, we also want to add .5em of padding so the top-most items are not hitting the left edge of the border.

I suppose I could do something like this, to improve readability:

```typescript
const outterPadding = 0.5;
const indentPadding = 1.3;
const paddingLeft = (outterPadding + (indentLevel * indentPadding)) + 'em';
const style = { paddingLeft };
```

[the remark this answers is not in the document]
Comment (jernestmyers) [reply packages/libs/eda/src/lib/map/analysis/MapSideNavigation.tsx:27]: Oh no, I think it's readable and makes sense as-is. I just wasn't sure how you settled on the numbers used in the calculation!

[the remark this answers is not in the document]
Comment (dmfalke) [reply packages/libs/eda/src/lib/map/analysis/MapSideNavigation.tsx:27]: Ah, sorry! Yes, that was just trial and error :smile:

**Pull request merged**